testsuite/cluster-test/src/cluster.rs

```diff
@@ -213,4 +213,14 @@ impl Cluster {
     pub fn find_instance_by_pod(&self, pod: &str) -> Option<&Instance> {
         self.all_instances().find(|i| i.peer_name() == pod)
     }
+
+    pub fn instances_for_group(&self, validator_group: ValidatorGroup) -> Vec<Instance> {
```

Comment:
Can we change return type from Vec<Instance> to `impl Iterator<Item = &Instance>` and use smth like `self.all_instances().filter(...)`